planning/behavior_path_planner/autoware_behavior_path_start_planner_module/src/clothoid_pull_out.cpp

```diff
@@ -0,0 +1,1680 @@
+    point.longitudinal_velocity_mps = velocity;
+    point.lateral_velocity_mps = 0.0;
+    point.acceleration_mps2 = 0.0;
+    point.heading_rate_rps = 0.0;
+    point.front_wheel_angle_rad = 0.0;
+    point.rear_wheel_angle_rad = 0.0;
+
+    // Set time
+    if (i == 0) {
+      point.time_from_start.sec = 0;
+      point.time_from_start.nanosec = 0;
+    } else {
+      const double distance = std::sqrt(
+        std::pow(path_points[i].first - path_points[i - 1].first, 2) +
+        std::pow(path_points[i].second - path_points[i - 1].second, 2));
+      const double time_diff = distance / velocity;
+
+      // Use builtin_interfaces::msg::Duration
+      const auto prev_time = trajectory.points[i - 1].time_from_start;
+      const auto time_diff_sec = static_cast<int32_t>(time_diff);
+      const auto time_diff_nanosec = static_cast<uint32_t>((time_diff - time_diff_sec) * 1e9);
+
+      point.time_from_start.sec = prev_time.sec + time_diff_sec;
+      point.time_from_start.nanosec = prev_time.nanosec + time_diff_nanosec;
+
+      // Handle nanosecond overflow
+      if (point.time_from_start.nanosec >= 1000000000) {
+        point.time_from_start.sec += 1;
+        point.time_from_start.nanosec -= 1000000000;
+      }
+    }
+
+    trajectory.points.push_back(point);
+  }
+
+  return trajectory;
+}
+
+ClothoidPullOut::ClothoidPullOut(
+  rclcpp::Node & node, const StartPlannerParameters & parameters,
+  std::shared_ptr<autoware_utils::TimeKeeper> time_keeper)
+: PullOutPlannerBase{node, parameters, time_keeper}
+{
+  autoware::boundary_departure_checker::Param boundary_departure_checker_params;
+  boundary_departure_checker_params.footprint_extra_margin =
+    parameters.lane_departure_check_expansion_margin;
+  boundary_departure_checker_ =
+    std::make_shared<autoware::boundary_departure_checker::BoundaryDepartureChecker>(
+      boundary_departure_checker_params, vehicle_info_, time_keeper_);
+}
+
+std::optional<PullOutPath> ClothoidPullOut::plan(
+  const Pose & start_pose, const Pose & /*goal_pose*/,
+  const std::shared_ptr<const PlannerData> & planner_data, PlannerDebugData & planner_debug_data)
+{
+  // =====================================================================
+  // STEP 1: Parameter setting and initialization
+  // =====================================================================
+  const double initial_velocity = parameters_.clothoid_initial_velocity;
+  const double acceleration = parameters_.clothoid_acceleration;
+  const std::vector<double> max_steer_angle_degs = parameters_.clothoid_max_steer_angle_degs;
+
+  // Convert degrees to radians from parameters
+  std::vector<double> max_steer_angle;
+  for (const auto & deg : max_steer_angle_degs) {
+    max_steer_angle.push_back(deg * M_PI / 180.0);
+  }
+
+  const double max_steer_angle_rate_deg_per_sec =
+    parameters_.clothoid_max_steer_angle_rate_deg_per_sec;
+  const double max_steer_angle_rate = max_steer_angle_rate_deg_per_sec * M_PI / 180.0;
```

Comment:
autoware_utils::deg2rad

Reply:
addressed in https://github.com/autowarefoundation/autoware_universe/pull/11040/commits/5a18805ba0ed36c139f24f3d319eac7df743f6b3